src/adaptors/yield-yak/index.js

```diff
@@ -0,0 +1,31 @@
+const axios = require('axios');
+
+const main = async () => {
+  const [{ data: farms }, { data: apys }] = await Promise.all([
+    axios.get('https://staging-api.yieldyak.com/farms'),
+    axios.get('https://staging-api.yieldyak.com/apys'),
+  ]);
+
+  return farms
+    .filter((farm) => apys.hasOwnProperty(farm.address))
+    .map((farm) => {
+      return {
+        pool: farm.address,
+        chain: 'Avalanche',
+        project: 'yield-yak',
+        symbol: farm.name,
+        apy: apys[farm.address].apy,
+        rewardTokens: [farm.rewardToken.address],
+        underlyingTokens: farm.depositToken.underlying,
+        // this is in terms of deposit tokens, but there is no simple way to convert this
+        // to USD, so for now just adding this. if ever this gets fixed, the graph should still
+        // look some kind of proportionally scaled version of the original?
+        tvlUsd: +farm.totalDeposits,
```

Comment:
This can be quite misleading though lol

Reply:
Hello, thank you for your PR. 
You should convert token tvl to usd tvl using coingecko price api or defillama price api (see here https://defillama.com/docs/api)

Reply:
not an easy or simple thing to do for LP tokens or vaults with multiple underlying tokens, etc.

Reply:
I am talking with the team to add token price or tvl directly in the API response

Reply:
> not an easy or simple thing to do for LP tokens or vaults with multiple underlying tokens, etc.

gm, ty for adding! we have many instances (such as uniswap) where we derive tvl usd values from underlying token balances of an LP position. so as long you can provide token addresses we can just use our own price api to calculate the usd values in a similar way

Reply:
> I am talking with the team to add token price or tvl directly in the API response

that works too :)

Reply:
> gm, ty for adding! we have many instances (such as uniswap) where we derive tvl usd values from underlying token balances of an LP position. so as long you can provide token addresses we can just use our own price api to calculate the usd values in a similar way

how can i use this pricing? i see tvl is accepted directly in usd terms, is there any option for providing just token balances and the driver uses price and token balance to calculate tvl?

Reply:
not yet! so on an adaptor level we expect tvl to be in usd values.

- for some projects (which share subgraph structure) we use utils.tvl to get from balances to usd values (it uses our own price api for doing that).
- If you need more fine grained control for your specific integration, you can just take a look at one of many examples in the adaptors folder, such as balancer: https://github.com/DefiLlama/yield-server/blob/master/src/adaptors/balancer/index.js#L299. you could also take a look at our official api docs which gives a testable example for that endpoint: https://defillama.com/docs/api

Lmk if anything unclear, we gonna figure it out sir!